modules/orchestrator/ref-best-practices-for-creating-workflows.adoc

```diff
@@ -0,0 +1,91 @@
++
+** Use imperative verbs such as `Call`, `Emit`, `For`, `Fork`, `Raise`, `Run`, `Set`, `Switch`, and `Wait`. These simple, universally understood terms make your workflow simple to read and understand.
+
+Structure and Extensibility:::
++
+** Use implicit default behaviors to reduce redundancy.
+** Declare components inline if they are not reusable to keep the definition self-contained.
+** Use external references to import and reuse shared components, which promotes a modular design.
+** Prioritize flexibility over strict enumerations to ensure extensibility and adaptability across different runtime environments.
+
+Data flow and runtime management::
++
+Controlling data flow is critical for efficient workflows. Tasks are the fundamental computing units of a workflow. The Domain Specific Language defines several default task types that runtimes must do. These include `Do`, `Listen`, `Raise`, `Run`, `Try`, and `Wait`.
++
+You can apply data transformations at several key points:
+
+Workflow input validation::: Before the workflow starts, validate input data against the `input.schema` and use the top-level `input.from` expression to filter out data that is not relevant data to the workflow context.
+
+Task Input/Output::: Before a task runs, transform its input using the `input.from` to match specific task requirements. After the task completes, use `output.as` runtime expression to transform its output before it is passed to the next task.
+
+Context Management::: Use the `export.as` runtime expression to update the context of the workflow. This evaluates the transformed task output, making sure the final output is accurate and relevant.
+
+Workflow output transformation::: Finally, transform the overall workflow output using `output.as` before returning it to the caller, making sure the final output is relevant.
+
+Security and error handling::
++
+Secrets::: Use Secrets with caution. Use them within the `input.from` runtime expression to prevent them from being unintentionally exposed. Avoid passing them directly in expressions or call inputs as this might expose sensitive information.
+
+Fault tolerance and error handling::: Serverless Workflow is designed with resilience in mind to recover from failures.
++
+Standard Errors:::: Use _standard error types_, such as timeouts, to categorize errors consistently across different runtimes. These also facilitate seamless migration from one runtime environment to another.
+Describe errors using the _Problem Details Request for Comments (RFC)_, with the `instance` property pointing to the component causing the error.
+
+Retries:::: Use `try` blocks to implement retries for transient errors, such as a temporary network issue.
+
+Timeouts:::: Configure timeouts for workflows and individual tasks. If a timeout occurs, you must interrupt the execution, and a standard timeout error with the type `https://serverlessworkflow.io/spec/0.8/errors/timeout` and status `408`is raised.
```

Comment:
the link gives a 404, https://github.com/serverlessworkflow/specification/blob/0.8.x/specification.md is working